clients/include/testing_gemv_batched.hpp

```diff
@@ -0,0 +1,307 @@
+        }
+    }
+
+    if(arg.timing)
+    {
+        int number_cold_calls = 2;
+        int number_hot_calls  = 100;
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
+
+        for(int iter = 0; iter < number_cold_calls; iter++)
+        {
+            rocblas_gemv_batched<T>(handle,
+                                    transA,
+                                    M,
+                                    N,
+                                    &h_alpha,
+                                    dAA,
+                                    lda,
+                                    dxA,
+                                    incx,
+                                    &h_beta,
+                                    dy_1A,
+                                    incy,
+                                    batch_count);
+        }
+
+        gpu_time_used = get_time_us(); // in microseconds
+
+        for(int iter = 0; iter < number_hot_calls; iter++)
+        {
+            rocblas_gemv_batched<T>(handle,
+                                    transA,
+                                    M,
+                                    N,
+                                    &h_alpha,
+                                    dAA,
+                                    lda,
+                                    dxA,
+                                    incx,
+                                    &h_beta,
+                                    dy_1A,
+                                    incy,
+                                    batch_count);
+        }
+
+        gpu_time_used     = (get_time_us() - gpu_time_used) / number_hot_calls;
+        rocblas_gflops    = batch_count * gemv_gflop_count<T>(M, N) / gpu_time_used * 1e6;
+        rocblas_bandwidth = batch_count * (1.0 * M * N) * sizeof(T) / gpu_time_used / 1e3;
+
+        // only norm_check return an norm error, unit check won't return anything
+        std::cout << "M,N,alpha,lda,incx,beta,incy,rocblas-Gflops,rocblas-GB/s,";
+        if(arg.norm_check)
+        {
+            std::cout << "CPU-Gflops,norm_error_host_ptr,norm_error_device_ptr";
+        }
+        std::cout << std::endl;
+
+        std::cout << M << "," << N << "," << h_alpha << "," << lda << "," << incx << "," << h_beta
+                  << "," << incy << "," << rocblas_gflops << "," << rocblas_bandwidth << ",";
+
+        if(arg.norm_check)
+        {
+            std::cout << cblas_gflops << ',';
+            std::cout << rocblas_error_1 << ',' << rocblas_error_2;
+        }
+
+        std::cout << std::endl;
+    }
+
+    for(int b = 0; b < batch_count; ++b)
+    {
+        CHECK_HIP_ERROR(hipFree(AA[b]));
+        CHECK_HIP_ERROR(hipFree(xA[b]));
+        CHECK_HIP_ERROR(hipFree(y_1A[b]));
+        CHECK_HIP_ERROR(hipFree(y_2A[b]));
+    }
+    CHECK_HIP_ERROR(hipFree(dAA));
+    CHECK_HIP_ERROR(hipFree(dxA));
+    CHECK_HIP_ERROR(hipFree(dy_1A));
+    CHECK_HIP_ERROR(hipFree(dy_2A));
```

Comment:
`device_vector` makes this unnecessary.

Reply:
See comment above

Reply:
Under refactoring as per discussion

Reply:
New device_batch_vector class added to rocblas_vector.hpp